_ml-commons-plugin/semantic-search.md

```diff
@@ -0,0 +1,1045 @@
+---
+layout: default
+title: Semantic search
+has_children: false
+nav_order: 140
+---
+
+# Semantic search
+
+By default, OpenSearch calculates document scores using the [Okapi BM25](https://en.wikipedia.org/wiki/Okapi_BM25) algorithm. BM25 is a keyword-based algorithm that performs well on a query containing keywords but fails to capture the semantic meaning of the query terms. Semantic search, unlike keyword-based search, takes into account the meaning of the query in the search context. Thus, semantic search performs well when a query requires natural language understanding. 
+
+In this tutorial, you'll learn how to:
+
+- Implement semantic search in OpenSearch.
+- Combine semantic search with keyword search to improve search relevance. 
+
+## Terminology
+
+It's helpful to understand the following terms before starting this tutorial:
+
+- _Semantic search_: Employs neural search in order to determine the intention of the user's query in the search context and improve search relevance. 
+- _Neural search_: When indexing documents containing text, neural search uses language models to generate vector embeddings from that text. When you then use a _neural query_, the query text is passed through a language model and the resulting vector embeddings are compared with the document text vector embeddings to find the most relevant results.
+
+## OpenSearch components for semantic search
+
+For this tutorial, you'll implement semantic search using the following OpenSearch components:
+
+- [Model group]({{site.url}}{{site.baseurl}}/ml-commons-plugin/model-access-control#model-groups)
+- [Pretrained language models provided by OpenSearch]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/)
+- [Ingest pipeline]({{site.url}}{{site.baseurl}}/api-reference/ingest-apis/index/)
+- [k-NN vector]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/knn-vector/)
+- [Neural search]({{site.url}}{{site.baseurl}}/search-plugins/neural-search/)
+- [Search pipeline]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/index/)
+- [Normalization processor]({{site.url}}{{site.baseurl}}/search-plugins/search-pipelines/normalization-processor/)
+- [Hybrid query]({{site.url}}{{site.baseurl}}/query-dsl/compound/hybrid/)
+
+You'll find the explanations of all these components as you follow the tutorial, so don't worry if you're not familiar with some of them. Each link in the preceding list will take you to a corresponding documentation section if you want to learn more.
+
+## Prerequisites
+
+Before you get started, make sure to update the default [cluster settings]({{site.url}}{{site.baseurl}}/api-reference/cluster-api/cluster-settings/) to the following:
+
+```json
+PUT _cluster/settings
+{
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "only_run_on_ml_node": "false",
+        "model_access_control_enabled": "true",
+        "native_memory_threshold": "99",
+        "allow_registering_model_via_url": "true"
```

Comment:
None of those settings are required for hybrid search to work, mainly they are for ml-commons to upload the model that we're using to create embeddings, in other words it's possible to make it run and not set any of that settings. Say if you're using one of pre-built models which makes sense for the simple example, then "allow_registering_model_via_url" is not needed at all. "native_memory_threshold" is needed to minimize chance of open circuit breaker when machine for node is not that powerful, "only_run_on_ml_node" is used if this is simple cluster without dedicated ml nodes, and nodes of other types can be used. "model_access_control_enabled" is also optional, I think if security plugin is enabled then it's not required. Maybe say that is one of simplest config that will work in local, and also put a reference to ml-common config docs

Reply:
Thanks, reworded.